### ISSUE TITLE
No subtitles downloaded during automatic process

### ISSUE THREAD

**Issue by codefaux** [morpheus65535/bazarr#1939]
**Describe the bug**
During scheduled processing, Bazaar seemingly does not download any subtitles, ever, and makes no indication of why. Downloads triggered via web UI from within Wanted list -- both Quick and Interactive -- work perfectly. No errors logged*.

**To Reproduce**
- OpenSubtitles is only subtitle provider
- Try to use software as correctly as aware
- Get nowhere because Bazarr silently ignores series without previously set Language
- Apply Language sets, get webUI-triggered Quick and Interactive subtitle downloads working
- Wait for scheduled tasks to run

**Expected behavior**
- Subtitles download periodically, and Wanted number decreases over time

**Actual behavior**
- Bazarr runs scheduled tasks, they take as long as they should(?), with Download tasks seemingly doing nothing, with no indication as to why, but all tasks claiming to complete successfully.

**Software**
 - Bazarr: 1.1.0
 - Radarr version: 4.1.0.6175
 - Sonarr version: 3.0.9.1549
 - OS: Linux

**Additional context**
First issue - THERE IS ABSOLUTELY ZERO USER-FACING INDICATION OF A PROBLEM. Seriously!? It just says it's doing the thing, ZERO related information provided in the logs, no feedback in any of the status pages, the Providers page which is INTENDED TO LIST ISSUES WITH PROVIDERS says "good". GOOD.

Second issue - Even with a deep dive into the logs, nothing of merit is provided until turning on the Debug Logging option. Within six or so seconds I had 50 pages of data to sift through, triple-spaced, 50 entries at a time, with as few as two actual words per entry. I know logging is a bit rough to manage on a UI, but WOW that's unfriendly.

Third issue - The actual core of the problem, intermixed about once per page, is a message that could have triggered a flag on the Providers list, "Skipping discarded provider 'opensubtitles'" *(PS WHY does that copy as a complete HTML document? I should've pasted it how it copied to highlight why that's a bad design)

Skipping discarded provider 'opensubtitles'
--- WHY was it discarded?

I can't just read the log - it's not IN the log. I'm NOT going to enable Debug logging and restart the service hoping to fish for an answer, it took literally a handful of seconds to write 500K in logs mid-run, and I don't know when the information I need is going to be provided.

**Comment by codefaux** [comment]
Ah - PS - I can MANUALLY trigger downloads from the Wanted list, _while the <html>
<body>


Search for wanted Series Subtitles |  
-- | --


<br class="Apple-interchange-newline">
</body>
</html> task is running_ (this time I'm not correcting it, you set it to copy that way, you deal with it) and the manual search will work, and decrement the Wanted number, but the automated task continues to not download files, while still performing disk IO and utilizing processor time, thus indicating it's actually trying to do things. What in the world is happening?

**Comment by morpheus65535** [comment]
Dude, being aware of your particular condition, I will take time to respond to this issue even if nothing here make me wanting to help you. 

Please provide your debug log and I'll look into it. This is probably a duplicate from #1919 as you already spotted. I've never run into that issue myself, I can't reproduce it. I don't say it doesn't exists, just that I can't debug it if I can't reproduce it.

**Comment by codefaux** [comment]
Nobody deserves to need to work directly with me.

More importantly, I apologize for my apparent tone. Consider my tone now quiet and level. In general I'm emotionless and fact-and-logic-oriented, but emphasis makes it seem like I'm scolding or lecturing. I don't intend that, I just mean to be clear and thorough behind where I'm coming from and why, because I feel like people disregard my concerns and thoughts too often I guess. Kind of like being told there is no problem when asking for configurable scheduling because a program loops on itself and brings my system to a grind. Then seeing later that others also complained and the exact thing I had requested had been implemented. So, I also came into this somewhat jaded. Again, I'm not trying to scold, just explaining myself. I know I'm in the wrong, and again, I apologize. It's exasperating to deal with an issue, to find out it's a known thing, which has logging and logic implemented, but -everything- is hidden behind debug logging. It didn't even warn me when I hadn't set a language per-series/movie so nothing could ever happen.

I got frustrated because you clearly carefully wrote logic to handle a condition which would trigger a provider to be unused, also carefully wrote a LOT of logging, but the way you use it is very "after the fact" and "behind closed doors" to the user, in software centered around expanding accessibility to media. I'm not justifying my actions; merely explaining why. I was in the wrong and I acknowledge that. I respect you, and it's obvious you're a smart person. I need to do better.

Back on topic.

Regarding discarded providers: We can agree it's a known issue. My respectful suggestion would be to elevate the logging surrounding "discarding" subtitle providers and handling "discarded" subtitle providers to Info or Error level, so it shows up without needing interaction with users to obtain -- and also far fewer privacy concerns. I don't know what's in the logs without reading them, and there are a -lot- of Debug level logs. I don't want the public to so much as see my directory structure, among other potential things which could be revealed. It's that kind of world. Info/Error level log facilities are there for noteworthy events which impact function, and since there aren't pages of them at a time it's easy to correlate and narrow the time range to log events with problems, for even a middling skill user. Even if a user has to clean them, that's a "send with the New Issue" and reduces your workload as a developer, and forced interactions with undesirables like myself.

Seeing the logs raised around the provider will probably reveal the problem as user error or at least indicate a cause. Good luck moving forward.
